### DISCUSSION TITLE
Maximum limit for batch in getFullList and getList?

### DISCUSSION THREAD

**Discussion by Salman9000** [pocketbase/pocketbase#6169] [Q&A] [answered]
What is the maximum number I can put in batch when doing getFullList?
`const resultList = await pb.collection<IExpandedMessageAndRepliedList>('messages').getFullList({
      filter: filterCondition,
      expand: 'campaign,user,user.replied,template',
      batch: someNumber
    });`
    
 Same goes for `getList(page, perPage)`. What can be the max value of perPage?

**Comment by ganigeorgiev** [comment, marked as answer]
In <= v0.22 was 500, and in v0.23+ is bumped to max 1000 because many users wanted bigger limit (_still not sure whether it was a good idea and may change in the future_).

Submitting a value higher than the limit will be normalized/capped.

The JS SDK by default uses 500 and in general I'd advice to leave it unchanged (_requesting 1k records with multiple expands in a single request may be less efficient and more resource intensive than sending several smaller requests_).

**Reply by Salman9000** [reply]
So how would i query records from a collection which can have somewhere between 1000 to 5000 records. I cannot know ahead of time how many records it will have, Using getFullList with batch value 500, it will get all records in a single request assuming i have 5000 records? Or I will have to query 10 times? 
Thanks

**Reply by ganigeorgiev** [reply]
Without details about your use case and what you are trying to do, I cannot help you further.

For me personally the question is not "how" but "why". Or in other words, what do you plan to do with fetching 5000 records? Rendering them all at once?

In any case, having a retrieval method that can return unlimited number of records in a single request is rarely a good idea and there are no plans to introduce other methods for now. If you want to return all records, use `getFullList()` with batch size based on what you think it is suited for your use case.
